### ISSUE TITLE
InvalidJavadocPosition: false positive for the generic constructor without access modifier

### ISSUE THREAD

**Issue by alx-ut** [checkstyle/checkstyle#11605]
Check documentation: https://checkstyle.sourceforge.io/config_javadoc.html#InvalidJavadocPosition

/var/tmp $ javac Test.java

/var/tmp $ cat Test.java
```
public class Test {
    /**
    * Test       
    */
    <E extends Enum> Test(E a) {
    }
}
```

/var/tmp $ cat config.xml
```
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="InvalidJavadocPosition"/>
    </module>
</module>

```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
````
Starting audit...                                                                                                                                                                                       
[ERROR] C:\Projects\checkstyle\Test.java:2:5: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]                                                                                 
Audit done.                                                                                                                                                                                             
Checkstyle ends with 1 errors.            
````

Expected: no validation error

**Comment by Shaman20** [comment]
Hey is this issue still valid and Can I work on this?
If yes please point me to resources that might help in solving this.

**Comment by nick-mancuso** [comment]
> Hey is this issue still valid and Can I work on this?
> If yes please point me to resources that might help in solving this.

1. Create input file containing given input in issue description.
2. Expand test input by adding classes with different constructor syntax, such as multiple generics, regular (not generic) constructors, different access modifiers on constructors , etc. 
3. Create new test in related check test file that uses your new input file.
4. Test should be failing, so begin debugging check, find and fix failure.

**Comment by Shaman20** [comment]
I am on it

**Comment by stoyanK7** [comment]
@Shaman20 Are you still working on it? If not, I want to take it.

**Comment by romani** [comment]
@stoyanK7, please work on it

**Comment by rnveach** [comment]
Fix was merged
